// @angular/forms/src/directives/select_multiple_control_value_accessor.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
"use strict";
var import0 = require('@angular/forms/src/directives/select_multiple_control_value_accessor');
var import1 = require('@angular/core/src/change_detection/change_detection');
var import2 = require('@angular/core/src/linker/view_utils');
var Wrapper_NgSelectMultipleOption = (function () {
    function Wrapper_NgSelectMultipleOption(p0, p1, p2) {
        this.changed = false;
        this.context = new import0.NgSelectMultipleOption(p0, p1, p2);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
    }
    Wrapper_NgSelectMultipleOption.prototype.check_ngValue = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this.changed = true;
            this.context.ngValue = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_NgSelectMultipleOption.prototype.check_value = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this.changed = true;
            this.context.value = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_NgSelectMultipleOption.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    Wrapper_NgSelectMultipleOption.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_NgSelectMultipleOption;
}());
exports.Wrapper_NgSelectMultipleOption = Wrapper_NgSelectMultipleOption;
var Wrapper_SelectMultipleControlValueAccessor = (function () {
    function Wrapper_SelectMultipleControlValueAccessor(p0, p1) {
        this.changed = false;
        this.context = new import0.SelectMultipleControlValueAccessor(p0, p1);
    }
    Wrapper_SelectMultipleControlValueAccessor.prototype.detectChangesInInputProps = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    Wrapper_SelectMultipleControlValueAccessor.prototype.detectChangesInHostProps = function (view, el, throwOnChange) {
    };
    return Wrapper_SelectMultipleControlValueAccessor;
}());
exports.Wrapper_SelectMultipleControlValueAccessor = Wrapper_SelectMultipleControlValueAccessor;
